Runtime/Device/ButtonRing/AdxIO4ButtonRing.cs

```diff
@@ -158,32 +155,48 @@ public override void ReadData(ReadOnlySpan<byte> data)
                 BA8  -> byte 4, position 4 OFF
                 Select -> byte 4, position 6 ON
             */
-            if (_currentState[4] != data[3])

```

Comment:
why remove these checks? If the _currentState byte is the same as the data in the packet then there is not change to be made so its wasted computations